diff --git a/Android.libv8.mk b/Android.libv8.mk
index 25691a6..88bc0c3 100644
--- a/Android.libv8.mk
+++ b/Android.libv8.mk
@@ -4,7 +4,7 @@ LOCAL_PATH := $(call my-dir)
 # ===================================================
 include $(CLEAR_VARS)
 
-include external/stlport/libstlport.mk
+include $(LOCAL_PATH)/../stlport/libstlport.mk
 
 ifeq ($(TARGET_ARCH),mips)
        LOCAL_MIPS_MODE=mips
@@ -27,6 +27,8 @@ V8_LOCAL_SRC_FILES += \
   src/platform-linux.cc \
   src/platform-posix.cc
 
+LOCAL_LDLIBS += -llog
+
 ifeq ($(TARGET_ARCH),x86)
 V8_LOCAL_SRC_FILES += src/atomicops_internals_x86_gcc.cc
 endif
@@ -66,11 +68,9 @@ LOCAL_GENERATED_SOURCES += $(V8_GENERATED_LIBRARIES)
 # Generate snapshot.cc
 ifeq ($(ENABLE_V8_SNAPSHOT),true)
 SNAP_GEN := $(intermediates)/snapshot.cc
-MKSNAPSHOT := $(HOST_OUT_EXECUTABLES)/mksnapshot.$(TARGET_ARCH)
-$(SNAP_GEN): PRIVATE_CUSTOM_TOOL = $(MKSNAPSHOT) --logfile $(intermediates)/v8.log $(SNAP_GEN)
-$(SNAP_GEN): $(MKSNAPSHOT)
-	$(transform-generated-source)
-LOCAL_GENERATED_SOURCES += $(SNAP_GEN)
+$(SNAP_GEN):
+	@cp -fp $(LOCAL_PATH)/../../snapshot.cc $(intermediates)/snapshot.cc
+LOCAL_SRC_FILES += src/snapshot.cc
 else
 LOCAL_SRC_FILES += \
   src/snapshot-empty.cc
diff --git a/Android.v8common.mk b/Android.v8common.mk
index 4c77a90..6a8ad9a 100644
--- a/Android.v8common.mk
+++ b/Android.v8common.mk
@@ -98,7 +98,9 @@ V8_LOCAL_SRC_FILES := \
 	src/v8threads.cc \
 	src/variables.cc \
 	src/version.cc \
-	src/zone.cc
+	src/zone.cc \
+	src/libraries.cc \
+	src/experimental-libraries.cc		
 
 ifeq ($(TARGET_ARCH),arm)
   V8_LOCAL_SRC_FILES += \
